k2/csrc/fsa.cu

```diff
@@ -233,17 +233,24 @@ int32_t GetFsaBasicProperties(const Fsa &fsa) {
   return ans;
 }
 
-Fsa FsaFromArray1(Array1<Arc> &array, bool *error) {
+Fsa FsaFromArray1(Array1<Arc> &array, int32_t num_states, bool *error) {
   NVTX_RANGE(K2_FUNC);
   const Arc *arcs_data = array.Data();
   ContextPtr &c = array.Context();
   int32_t num_arcs = array.Dim();
-  // We choose to return an Fsa with no states and no arcs.  We could also have
-  // chosen to return an Fsa with 2 states and no arcs.
-  if (num_arcs == 0)
-    return Fsa(EmptyRaggedShape(c, 2), Array1<Arc>(c, 0));
   *error = false;
 
+  if (num_states <= 0) {
+    K2_LOG(WARNING) << "Could not convert Array1 to FSA, "
+                       "there are no states";
+    *error = true;
+    return Fsa();
+  }
+
+  if (num_arcs == 0) {
+    return Fsa(EmptyRaggedShape(c, num_states), Array1<Arc>(c, 0));
```

Comment:
Here, I'm not sure about how to deal with the super final state. Should it be added in this case (num_arcs == 0)?

Reply:
No, if there is no start state there should be no super-final state (0 or 2 states are allowed, but not 1 state).